### PR TITLE
Repeat Functionality

### PR DESCRIPTION
**Pull request by AleanaA**
After creating your pull request, tick these boxes if they are applicable to you.

- [x] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.5/3.6

----

### Description
Introduces Repeating queue functionality.
Current method to avoid errors, which could be done far better, is to check for data output from ffmpeg (No other method appeared viable as is, as there are any number of other errors outside the 404 error I used to test that could occur) and if data output is received, to turn off repeat. 
This means that the queue won't be lost, and the queue will begin to carry on as if repeat was turned off. If this occurs, users with permission to use the `repeat` command can easily turn it back on

### Related issues (if applicable)

Any repeat feature requests

### CODE REVIEW

File: musicbot/player.py [hunk 1]
```diff
@@ -304,6 +315,14 @@ def play(self, _continue=False):
 
                 stderr_thread.start()
 
+                reperrcatch = Thread(
+                    target=self.repeat_err,
+                    args=(self._current_player._player.source.original._process, self._stderr_future),
```
Comment (Chaos21) [inline musicbot/player.py:344]: When runned, it seems that there's an error at this line

**Comment by AleanaA** [comment]
Error being?

**Comment by Chaos21** [comment]
@AleanaA 
image

Here's the error I've got. I've placed correctly the modifications, I've even removed them and tried again, and again.

**Comment by AleanaA** [comment]
Can I have more information? What commands were run prior, what commands
were run after, are you on the review branch, etc.

**Comment by Chaos21** [comment]
@AleanaA Of course.
I've tested it many times.
So. When running, the bot isn't able to play music. All his commands work like he can join the voice channel, etc, but it's like the bot is in 'pause'. the current music is like not played as all. The green circle around the avatar don't appear.
I'm on the 1.9.7 version.

**Comment by Chaos21** [comment]
I don't know if when you're running the program, you'll get the same error. As well, it's what I've got.

**Comment by AleanaA** [comment]
All pull requests target review. Review uses d.py rewrite, and no pull
requests should be compatible with the master brach until the two get
merged. Please use review if you want this functionality to work.

**Review by maquikun — APPROVED** [review]
Works for me on the latest version of the bot, Got some unexpected indentation errors at first but fixed them buy copying and pasting lines 318 and 328.

**Comment by YuiyuD** [comment]
@AleanaA I got this error when using skip in list queue, seems, your repeat only repeat 1 song, in the list they will skip all
https://cdn.discordapp.com/attachments/549474261878964229/552680564688289803/unknown.png

File: musicbot/bot.py [hunk 2]
```diff
@@ -1941,6 +1941,19 @@ def check(reaction, user):
...
+        Usage:
+            {command_prefix}repeat
+            
+        Toggles playlist repeat.
+        """
+        self.config.repeat = not self.config.repeat
+        if self.config.repeat:
+            return Response('Repeat enabled!', delete_after=10)
+        else:
+            return Response('Repeat disabled!', delete_after=10)
```
Comment (jayktaylor) [inline musicbot/bot.py:2004]: Ternary op probably better here

File: musicbot/bot.py [hunk 3]
```diff
@@ -2224,6 +2237,11 @@ def check(reaction, user):
             lines.append(
                 self.str.get('cmd-queue-none', 'There are no songs queued! Queue something with {}play.').format(self.config.command_prefix))
 
+        if self.config.repeat:
+            lines.append("Repeat mode is **on**")
+        else:
+            lines.append("Repeat mode is **off**")
```
Comment (jayktaylor) [inline musicbot/bot.py:2293]: Another place for a ternary op

**Review by jayktaylor — CHANGES_REQUESTED** [review]
Can you ensure this works against the latest code in review?

**Comment by AleanaA** [comment]
Unmaintained, rewrite of this PR planned.

**Pull request closed without merging**